libevm/EVMC.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <evmjit.h>
+#include <libevm/VMFace.h>
+
+// FIXME: Move to EVM-C evm.h.
+typedef struct evm_instance* (*evm_create_fn)(void);
+
+namespace dev
+{
+namespace eth
+{
+class EVM;
+
+/// Translate the EVMSchedule to EVM-C revision.
+evm_revision toRevision(EVMSchedule const& _schedule);
+
+template<evm_create_fn createFn>
+class EVMC : public VMFace
+{
+public:
+    owning_bytes_ref exec(u256& io_gas, ExtVMFace& _ext, OnOpFunc const& _onOp) final;
+
+    bool isCodeReady(evm_revision _mode, uint32_t _flags, h256 _codeHash);
+    void compile(evm_revision _mode, uint32_t _flags, bytesConstRef _code, h256 _codeHash);
+
+private:
+    static EVM& instance();
```

Comment:
This shouldn't be static then

Reply:
So it would be something like
```
std::unique_ptr<EVMC> createEvmJit()
{
  return return std::unique_ptr<VMFace>(new EVMC{evmjit_create});
}

EVMC(evm_create_fn _createFn) : m_createFn(_createFn) {}

EVM& EVMC::instance()
{
  static EVM evm{m_createFn};
  return evm;
}
```
It's not a lot of changes and will be simpler I think

Reply:
Why not?

Reply:
I mean if `EVMC` would be not a template, that different instances of `EVMC` would need different `EVM` instance

Reply:
And if we want to create only once singleton-style, then we could make `EVMC` itself a singleton I guess

Reply:
Wait no, my example below won't work. How about something like this.
```
std::unique_ptr<EVMC> createEvmJit()
{
  static EVM evm{evmjit_create};
  return std::unique_ptr<VMFace>(new EVMC(evm));
}

EVMC(EVM& _evm) : m_instance(_evm) {}

EVM& EVMC::instance()
{
  return m_instance;
}
```